beacon-chain/forkchoice/doubly-linked-tree/store.go

```diff
@@ -145,8 +147,12 @@ func (s *Store) insert(ctx context.Context,
 
 // updateCheckpoints Update the justified / finalized epochs in store if necessary.
 func (s *Store) updateCheckpoints(justifiedEpoch, finalizedEpoch types.Epoch) {
-	s.justifiedEpoch = justifiedEpoch
-	s.finalizedEpoch = finalizedEpoch
+	if s.justifiedEpoch < justifiedEpoch {
+		s.justifiedEpoch = justifiedEpoch
+	}
+	if s.finalizedEpoch < finalizedEpoch {
+		s.finalizedEpoch = finalizedEpoch
+	}
```

Comment:
Do you want to log some warnings if someone attempts to update justified/finalized epochs less than current?